### PR TITLE
Add support for reading SQL files from S3

### PR DESCRIPTION
**Pull request by tym-xqo**
- add s3fs library
- `result_from_file()` can read from a S3 URL to a file
- `result_by_name()` will check for the name by globbing through a bucket if $QUERY_PATH is a bucket URL
- update dependencies

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,9 +40,18 @@ Passing argument `returns` to `db.result()` (or `result_from_file()`) overrides
...
 
-`result_by_name()` looks for SQL files in a local directory — `${PWD}/query_files` by default, or you may specify any arbitrary filesystem location by setting `$QUERY_PATH` in the environment. The `query_name` argument is the [stem](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem) of the desired file.
+- `result_by_name()` looks up files with a `.sql` extension in a local directory — it looks in `${PWD}/query_files` by default, or you may specify any arbitrary filesystem location by setting `$QUERY_PATH` in the environment. The `query_name` argument is the [stem](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem) of the desired file, i.e. the base name of the file without the `.sql` extension.
+
+#### Files from S3
+
+In addition to the local filesystem, `sqla-raw` can use [`s3fs`](https://s3fs.readthedocs.io/en/latest/) to read SQL files from an S3 bucket. It checks whether any path specifications are in the form of URLs that begin with the `s3://` scheme component.
```
Comment (jessebmiller) [inline README.md:49]: This is so nice!

**Review by jessebmiller — APPROVED** [review]
Seems pretty straight forward and nice

File: README.md (hunk 1 above)
Comment (tym-xqo) [reply README.md:49]: Thx!

File: raw/db.py [hunk 2]
```diff
@@ -128,36 +128,67 @@ def result(sql, returns="dict", autocommit=False, **kwargs):
...
-
-    # Read the given file into memory and pass to result().
-    with open(path) as f:
-        sql = f.read()
-        rows = result(sql=sql, returns=returns, **kwargs)
-        return rows
+    # If `path` is a S3 url, read it from there
+    if path.startswith("s3://"):
+        bucket = s3fs.S3FileSystem(anon=False)
+        with bucket.open(path, "rt") as f:
```
Comment (grymoire7) [inline raw/db.py:134]: ℹ️ The documentation for s3fs says that [only binary](https://s3fs.readthedocs.io/en/latest/index.html#limitations) mode is currently supported here.

> modestr
>     One of ‘rb’, ‘wb’, ‘ab’. These have the same meaning as they do for the built-in open function.

**Pull request merged**